### PR TITLE
buffs the radio jammer

### PR DESCRIPTION
**Pull request by roarbark**
## What Does This PR Do
radio jammer is a tiny item now, gets an action button to activate it, like flashlights for example.
new sprites, active and inactive, new inhands
some flavour and more precise examine texts

## Why It's Good For The Game
the effect of the radio jammer is actually pretty good. how come its still not picked often?

its overshadowed by all the full mutes we currently have, from emps to chems or abilities that instantly guarantee a silent kill; making the radio jammer pull up likewise isnt going to be good for the game, in my opinion the radio jam is exactly the kind of mute that all mutes in this game should be. any listeners know somethings up, but with a good probability not what exactly. there's a lead but no proof.

therefore, i try to cut back at its weaknesses and try to make it easier to handle- giving it an action button and making it fit pockets gives people the ability to selectively turn it on or off quicker, making it less bulky in your kit and less obvious that its in the game by random passerbys trying to comm.

## Images of changes

test1

test2

## Testing

https://user-images.githubusercontent.com/41054538/195653673-3cd8a7ec-d551-49cf-a8d7-ed7df3786d85.mp4

## Changelog
:cl:
tweak: radio jammer is now a small item, has an action button
imageadd: resprites the radio jammer
/:cl:

### CODE REVIEW

File: code/game/objects/items/devices/traitordevices.dm [hunk 1]
```diff
@@ -1,22 +1,35 @@
...
+	if(active)
+		icon_state = "[initial(icon_state)]-on"
+	else
+		icon_state = "[initial(icon_state)]"
+
 /obj/item/jammer/attack_self(mob/user)
-	to_chat(user, "<span class='notice'>You [active ? "deactivate" : "activate"] [src].</span>")
+	to_chat(user, "<span class='notice'>You [active ? "deactivate" : "activate"] [src]. [active ? "It goes quiet with a small click." : "It starts to hum silently."] </span>")
 	active = !active
+	update_icon()
```
Comment (Contrabang) [inline code/game/objects/items/devices/traitordevices.dm:25]:
```suggestion
	update_icon(UPDATE_ICON_STATE)
```
Just a slight change needed to reflect changes made by #17943

File: code/game/objects/items/devices/traitordevices.dm [hunk 2]
```diff
@@ -1,22 +1,35 @@
...
 
+/obj/item/jammer/update_icon_state()
+	if(active)
+		icon_state = "[initial(icon_state)]-on"
+	else
+		icon_state = "[initial(icon_state)]"
+
 /obj/item/jammer/attack_self(mob/user)
-	to_chat(user, "<span class='notice'>You [active ? "deactivate" : "activate"] [src].</span>")
+	to_chat(user, "<span class='notice'>You [active ? "deactivate" : "activate"] [src]. [active ? "It goes quiet with a small click." : "It starts to hum silently."] </span>")
```
Comment (Contrabang) [inline code/game/objects/items/devices/traitordevices.dm:23]: This is a bit awkward with src being surrounded by two of the same checks. Is there any way for this to be one `active ?` check?

File: code/game/objects/items/devices/traitordevices.dm [hunk 3]
```diff
@@ -1,22 +1,35 @@
 /obj/item/jammer
 	name = "radio jammer"
-	desc = "Device used to disrupt nearby radio communication."
+	desc = "Fog of war that fits your pocket. Flicking the switch and extending the antenna will scramble nearby radio comms, making outgoing messages hard to understand."
```
Comment (Contrabang) [inline code/game/objects/items/devices/traitordevices.dm:3]: Description is a bit wordy, but thats just my opinion

**Review by Contrabang — CHANGES_REQUESTED** [review]

File: code/game/objects/items/devices/traitordevices.dm (hunk 3 above)
Comment (roarbark) [reply code/game/objects/items/devices/traitordevices.dm:3]: yeah i tried to make it more precise, and not concise. i think brevity in stuff like this hurts the theme a lot.

File: code/game/objects/items/devices/traitordevices.dm (hunk 1 above)
Comment (roarbark) [reply code/game/objects/items/devices/traitordevices.dm:25]: oh, sure let me see

Comment (roarbark) [reply code/game/objects/items/devices/traitordevices.dm]: i did this

File: code/game/objects/items/devices/traitordevices.dm (hunk 2 above)
Comment (roarbark) [reply code/game/objects/items/devices/traitordevices.dm]: i did this

File: code/game/objects/items/devices/traitordevices.dm [hunk 4]
```diff
@@ -1,22 +1,35 @@
...
-	to_chat(user, "<span class='notice'>You [active ? "deactivate" : "activate"] [src].</span>")
+	to_chat(user, "<span class='notice'>You [active ? "deactivate [src]. It goes quiet with a small click." : "activate [src]. It starts to hum silently."] </span>")
 	active = !active
+	update_icon(UPDATE_ICON_STATE)
 	if(active)
 		GLOB.active_jammers |= src
 	else
 		GLOB.active_jammers -= src
+	for(var/X in actions)
+		var/datum/action/A = X
```
Comment (farie82) [inline code/game/objects/items/devices/traitordevices.dm:31]:
```suggestion
	for(var/datum/action/item_action/toggle_radio_jammer/A in actions)
```
Wouldn't it make more sense to just update the jammer action?

File: code/game/objects/items/devices/traitordevices.dm [hunk 5]
```diff
@@ -1,22 +1,35 @@
...
 
+/obj/item/jammer/update_icon_state()
+	if(active)
+		icon_state = "[initial(icon_state)]-on"
+	else
+		icon_state = "[initial(icon_state)]"
+
 /obj/item/jammer/attack_self(mob/user)
-	to_chat(user, "<span class='notice'>You [active ? "deactivate" : "activate"] [src].</span>")
+	to_chat(user, "<span class='notice'>You [active ? "deactivate [src]. It goes quiet with a small click." : "activate [src]. It starts to hum silently."] </span>")
```
Comment (farie82) [inline code/game/objects/items/devices/traitordevices.dm:23]:
```suggestion
	to_chat(user, "<span class='notice'>You [active ? "deactivate [src]. It goes quiet with a small click." : "activate [src]. It starts to hum softly."] </span>")
```
How can something be audible yet silent :P
Another word would also work here

File: code/datums/action.dm [hunk 6]
```diff
@@ -388,6 +388,10 @@
 	if(istype(H))
 		H.toggle_geiger_counter()
 
+/datum/action/item_action/toggle_radio_jammer
+	name = "Toggle Radio Jammer"
+	desc = "Turns your Jammer on or off. Hush, you."
```
Comment (farie82) [inline code/datums/action.dm:393]:
```suggestion
	desc = "Turns your jammer on or off. Hush, you."
```

**Comment by Denthamos** [comment]
The sprite looks a bit flat to me but otherwise I like the performance changes of it.

File: code/game/objects/items/devices/traitordevices.dm (hunk 4 above)
Comment (roarbark) [reply code/game/objects/items/devices/traitordevices.dm:31]: i dont know, i just took the snippet from flashlights. if this is better, i'll test it though.

File: code/game/objects/items/devices/traitordevices.dm (hunk 5 above)
Comment (roarbark) [reply code/game/objects/items/devices/traitordevices.dm:23]: the idea was to present it as if you could feel the hum in your hand but not hear it. softly is a nice word though, i'm thinking about using that instead

File: code/datums/action.dm (hunk 6 above)
Comment (roarbark) [reply code/datums/action.dm:393]: oh, whoops

File: code/game/objects/items/devices/traitordevices.dm (hunk 4 above)
Comment (roarbark) [reply code/game/objects/items/devices/traitordevices.dm]: i did this

File: code/game/objects/items/devices/traitordevices.dm (hunk 5 above)
Comment (roarbark) [reply code/game/objects/items/devices/traitordevices.dm]: i did this

File: code/datums/action.dm (hunk 6 above)
Comment (roarbark) [reply code/datums/action.dm]: i did this

**Comment by roarbark** [comment]
> The sprite looks a bit flat to me but otherwise I like the performance changes of it.

its supposed to be a really smooth and slim item. i made it tiny afterall.

File: code/game/objects/items/devices/traitordevices.dm (hunk 3 above)
Comment (roarbark) [reply code/game/objects/items/devices/traitordevices.dm:3]: im gonna resolve this i think

**Review by hal9000PR — APPROVED** [review]

**Review by SteelSlayer — APPROVED** [review]
